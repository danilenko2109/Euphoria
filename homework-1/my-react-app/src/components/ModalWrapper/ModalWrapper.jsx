const ModalWrapper = ({ children, onClose }) => {
  return (
    <div className="modal-overlay" data-testid="modal-overlay" onClick={onClose}>
      <div
        className="modal"
        data-testid="modal-inner"
        onClick={(e) => {
          e.stopPropagation();
        }}
      >
        {children}
      </div>
    </div>
  );
};

export default ModalWrapper;
