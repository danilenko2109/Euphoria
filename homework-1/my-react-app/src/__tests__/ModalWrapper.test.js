import { render, screen, fireEvent } from '@testing-library/react';
import ModalWrapper from '../components/ModalWrapper/ModalWrapper';
import renderer from 'react-test-renderer'; 

/* global describe, test, expect, jest */



describe('ModalWrapper component', () => {
  test('renders children content', () => {
    render(
      <ModalWrapper onClose={() => {}}>
        <div>Modal content</div>
      </ModalWrapper>
    );

    expect(screen.getByText('Modal content')).toBeInTheDocument();
  });

  test('calls onClose when clicking overlay', () => {
    const onCloseMock = jest.fn();

    render(
      <ModalWrapper onClose={onCloseMock}>
        <div>Modal content</div>
      </ModalWrapper>
    );

    const overlay = screen.getByTestId('modal-overlay');
    fireEvent.click(overlay);

    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  test('does not call onClose when clicking inside modal content', () => {
    const onCloseMock = jest.fn();

    render(
      <ModalWrapper onClose={onCloseMock}>
        <div>Modal content</div>
      </ModalWrapper>
    );

    const modalInner = screen.getByTestId('modal-inner');
    fireEvent.click(modalInner);

    expect(onCloseMock).not.toHaveBeenCalled();
  });

  test('matches snapshot', () => {
    const tree = renderer
      .create(
        <ModalWrapper onClose={() => {}}>
          <div>Modal content</div>
        </ModalWrapper>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
