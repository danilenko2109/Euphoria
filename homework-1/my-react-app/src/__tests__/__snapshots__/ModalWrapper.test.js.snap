// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper component matches snapshot 1`] = `null`;
